### PR TITLE
[DI] fix perf issue with lazy autowire error messages

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32711
| License       | MIT
| Doc PR        | -

See linked issue.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Benefit confirmed in #32711, ready.

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php [hunk 1]
```diff
@@ -375,20 +376,22 @@ private function set(string $type, string $id)
 
     private function createTypeNotFoundMessageCallback(TypedReference $reference, $label)
     {
-        $container = new ContainerBuilder($this->container->getParameterBag());
-        $container->setAliases($this->container->getAliases());
-        $container->setDefinitions($this->container->getDefinitions());
-        $container->setResourceTracking(false);
+        if (null === $this->typesClone->container) {
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: the `container` property should be declared as a private property of the class. We don't want to make it a dynamic public property.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: see https://github.com/symfony/symfony/blob/9216cb75ac11db014262f2d7fe5cb77b0b54db59/src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php#L31

Comment (stof) [reply src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: ah, I missed the fact it was an inherited property. All good then.

**Review by stof — APPROVED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @nicolas-grekas.

**Pull request merged**